src/main/java/util/PiazzaLogger.java

```diff
@@ -186,84 +234,49 @@ private LoggerPayload getLoggerPayload() {
 		}
 		return loggerPayload;
 	}
-
+	
 	/**
-	 * Gets the Executor that manages the log threads.
-	 */
-	public ThreadPoolTaskExecutor getExecutor() {
-		return executor;
-	}
-
-	/**
-	 * Runnable that sends logs to the pz-logger component HTTP endpoint
+	 * Method to index the log payload into elasticsearch.
+	 * 
+	 * @param loggerPayload
+	 *            Syslog RFC 5424 standard log payload
+	 * @param logMessage
+	 *            actual log message
+	 * @param severity
+	 *            Syslog RFC-5424 Protocol severity codes
 	 */
-	public class LogWorker implements Runnable {
-		private LoggerPayload loggerPayload;
-		private String logMessage;
-		private Severity severity;
+	private void indexLog(LoggerPayload loggerPayload, String logMessage, Severity severity) {
+		// Setting generic fields on logger payload
+		loggerPayload.setSeverity(severity);
+		loggerPayload.setMessage(logMessage);
+		loggerPayload.setTimestamp(new DateTime());
 
-		public LogWorker(LoggerPayload loggerPayload, String logMessage, Severity severity) {
-			this.loggerPayload = loggerPayload;
-			this.logMessage = logMessage;
-			this.severity = severity;
+		// Log to console if requested
+		try {
+			if (logToConsole.booleanValue()) {
+				LOGGER.info(loggerPayload.toString());
+			}
+		} catch (Exception exception) { /* Do nothing. */
+			LOGGER.error("Could not log message to console. Application property is not set", exception);
 		}
 
-		public void run() {
-			sendLogs(loggerPayload, logMessage, severity);
+		// saving log to elastic search
+		String loggerPayloadJson = "";
+		try {
+			loggerPayloadJson = new ObjectMapper().writeValueAsString(loggerPayload);
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
```

Comment:
Handle this exception property. Shouldn't we do or say something if the payload fails to serialize? Also, log the exception object. SonarQube will complain.